### PR TITLE
SASS-3159 Implemented GET/PUT/DELETE for pension Income

### PR DESCRIPTION
**Pull request by abiodxn**
### Description
Added PUT/DELETE/GET for pension income 

Add a link to the relevant story in Jira
[SASS-3159](https://jira.tools.tax.service.gov.uk/browse/SASS-3159)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [ ]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

**Comment by erasmos-hmrc** [comment]
I've confirmed that [the dependencies](https://catalogue.tax.service.gov.uk/service/income-tax-pensions) are up to date.

**Comment by erasmos-hmrc** [comment]
Given that this is the backend, the journey tests are irrelevant so I didn't run them.

**Comment by erasmos-hmrc** [comment]
Note that I haven't run the journey tests yet.

File: app/connectors/PensionIncomeConnector.scala [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    def desCall(implicit hc: HeaderCarrier): Future[CreateOrAmendPensionIncomeResponse] = {
+      http.PUT[CreateUpdatePensionIncomeModel,
+        CreateOrAmendPensionIncomeResponse](incomeSourcesUri,
+        pensionIncome)(CreateUpdatePensionIncomeModel.format.writes, CreateOrAmendPensionIncomeHttpReads, hc, ec)
+    }
+
+    desCall(desHeaderCarrier(incomeSourcesUri))
+  }
+
+  def deletePensionIncome(nino: String, taxYear: Int)(implicit hc: HeaderCarrier): Future[DeletePensionIncomeResponse] = {
```
Comment (erasmos-hmrc) [inline app/connectors/PensionIncomeConnector.scala:62]: Why not validate before calling further?

**Comment by erasmos-hmrc** [comment]
I'd suggest that you mention the three new endpoints in the README.

It would also be good to give an example of the tax year format.

**Comment by erasmos-hmrc** [comment]
I had a chat with @abiodxn and we both wondered as to the purpose of our API (`income-tax-pensions`).

If it's meant to be a straight pass-through to the downstream service (like DES), then why do we sometimes have our own validation, for example: the schema validation for the PUT, but don't bother validating things like the tax year or the NINO.

And apparently we do have such validation in the frontend.

And If it's meant to be a straight pass-through, what value does it have? Why not just have `income-tax-pensions-frontend` call DES directly? It would save a network hop, not to mention the overhead of maintaining the service.

I'll seek some answers around this, including chatting with @GavinPetersonGit .

File: it/api/DeletePensionIncomeITest.scala [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+          .withHttpHeaders(mtditidHeader)
+          .delete()) {
+          result =>
+            result.status mustBe BAD_REQUEST
+            Json.parse(result.body) mustBe Json.obj(
+              "code" -> "INVALID_TAX_YEAR", "reason" -> "Submission has not passed validation. Invalid parameter taxYear.")
+        }
+      }
+
+      "return 400 if a there is an invalid taxable entity id (nino)" in new Setup {
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/api/DeletePensionIncomeITest.scala:79]: As agreed, we'll ignore this validation question for now, as we need to better understand the purpose of this backend (`income-tax-pensions`).

File: it/api/GetPensionIncomeITest.scala [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+        .withHttpHeaders(mtditidHeader)
+        .get) {
+          result =>
+            result.status mustBe BAD_REQUEST
+            Json.parse(result.body) mustBe Json.obj(
+              "code" -> "INVALID_TAX_YEAR", "reason" -> "Submission has not passed validation. Invalid parameter taxYear.")
+        }
+      }
+
+      "return 400 if a there is an invalid taxable entity id (nino)" in new Setup {
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/api/GetPensionIncomeITest.scala:107]: As agreed, we'll ignore this validation question for now, as we need to better understand the purpose of this backend (`income-tax-pensions`).

File: it/api/GetPensionIncomeITest.scala [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+        whenReady(buildClient(serviceUrl)
+          .withHttpHeaders(mtditidHeader)
+          .get) {
+          result =>
+            result.status mustBe OK
+            Json.parse(result.body) mustBe Json.parse(GetPensionIncomeDesResponseBody)
+        }
+      }
+
+      "return 400 if there is an invalid tax year" in new Setup {
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/api/GetPensionIncomeITest.scala:91]: As agreed, we'll ignore this validation question for now, as we need to better understand the purpose of this backend (`income-tax-pensions`).

File: app/connectors/PensionIncomeConnector.scala [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+
+class PensionIncomeConnector @Inject()(val http: HttpClient,
+                                       val appConfig: AppConfig)(implicit ec: ExecutionContext) extends DesConnector {
+
+  def pensionIncomeSourceUri(nino: String, taxYear: Int): String =
+    appConfig.desBaseUrl + s"/income-tax/income/pensions/$nino/${desTaxYearConverter(taxYear)}"
+
+  def getPensionIncome(nino: String, taxYear: Int)(implicit hc: HeaderCarrier): Future[GetPensionIncomeResponse] = {
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply app/connectors/PensionIncomeConnector.scala:36]: As agreed, we'll ignore this validation question for now, as we need to better understand the purpose of this backend (`income-tax-pensions`).

File: app/connectors/PensionIncomeConnector.scala [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    val incomeSourcesUri: String = pensionIncomeSourceUri(nino, taxYear)
+
+    def desCall(implicit hc: HeaderCarrier): Future[GetPensionIncomeResponse] = {
+      http.GET[GetPensionIncomeResponse](incomeSourcesUri)
+    }
+
+    desCall(desHeaderCarrier(incomeSourcesUri))
+  }
+
+  def createOrAmendPensionIncome(nino: String, taxYear: Int, pensionIncome: CreateUpdatePensionIncomeModel)
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply app/connectors/PensionIncomeConnector.scala:46]: As agreed, we'll ignore this validation question for now, as we need to better understand the purpose of this backend (`income-tax-pensions`).

File: it/api/DeletePensionIncomeITest.scala [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+        whenReady(buildClient(serviceUrl)
+          .withHttpHeaders(mtditidHeader)
+          .delete) {
+          result =>
+            result.status mustBe NO_CONTENT
+
+        }
+      }
+
+      "return 400 if a there is an invalid tax year" in new Setup {
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/api/DeletePensionIncomeITest.scala:60]: As agreed, we'll ignore this validation question for now, as we need to better understand the purpose of this backend (`income-tax-pensions`).

File: it/connectors/PensionIncomeConnectorISpec.scala [hunk 8]
```diff
@@ -0,0 +1,444 @@
...
+      val expectedResult = Json.parse(expectedResponseBody).as[GetPensionIncomeModel]
+      stubGetWithResponseBody(desUrl, OK, expectedResponseBody)
+
+      implicit val hc: HeaderCarrier = HeaderCarrier()
+      val result = await(connector.getPensionIncome(nino, taxYear)(hc)).right.get
+
+      result mustBe Some(expectedResult)
+    }
+
+    "return a Parsing error INTERNAL_SERVER_ERROR response" in {
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/connectors/PensionIncomeConnectorISpec.scala:155]: As agreed, it's probably enough for now just to have Wiremock expect the call, so as to remove the warning.

At times we may want to assert the auditing, but it's not yet clear.

File: test/models/GetPensionIncomeModelSpec.scala [hunk 9]
```diff
@@ -0,0 +1,276 @@
...
+      )
+    )
+
+    val minimumPensionIncomeModel: GetPensionIncomeModel = GetPensionIncomeModel(
+      submittedOn = "2022-07-28T07:59:39.041Z",
+      deletedOn = None,
+      foreignPension = Seq(
+        ForeignPension(
+          countryCode = "FRA",
+          taxableAmount = 1999.99,
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply test/models/GetPensionIncomeModelSpec.scala:191]: I confirmed with @abiodxn that this is indeed realistic.

File: test/models/GetPensionIncomeModelSpec.scala [hunk 10]
```diff
@@ -0,0 +1,276 @@
...
+          amountBeforeTax = None,
+          taxTakenOff = None,
+          specialWithholdingTax = None,
+          foreignTaxCreditRelief = None
+        )
+      ),
+      overseasPensionContribution = Seq(
+        OverseasPensionContribution(
+          customerReference = None,
+          exemptEmployersPensionContribs = 1999.99,
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply test/models/GetPensionIncomeModelSpec.scala:201]: I confirmed with @abiodxn that this is indeed realistic.

File: test/models/GetPensionIncomeModelSpec.scala [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+        )
+      )
+
+    val pensionIncomeWithMinForeignPensionModel: GetPensionIncomeModel = GetPensionIncomeModel(
+      submittedOn = "2022-07-28T07:59:39.041Z",
+      deletedOn = Some("2022-07-28T07:59:39.041Z"),
+      foreignPension = Seq(
+        ForeignPension(
+          countryCode = "FRA",
+          taxableAmount = 1999.99,
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply test/models/GetPensionIncomeModelSpec.scala:163]: I confirmed with @abiodxn that this is indeed realistic.

File: app/connectors/PensionIncomeConnector.scala (hunk 1 above)
Comment (erasmos-hmrc) [reply app/connectors/PensionIncomeConnector.scala:62]: As agreed, we'll ignore this validation question for now, as we need to better understand the purpose of this backend (`income-tax-pensions`).

File: it/connectors/PensionIncomeConnectorISpec.scala [hunk 12]
```diff
@@ -0,0 +1,444 @@
...
+    CreateUpdatePensionIncomeModel(fullForeignPensionModel, fullOverseasPensionContributionModel)
+  val fullCreateOrUpdatePensionIncomeJsonBody: String = Json.toJson(fullCreateOrUpdatePensionIncomeData).toString()
+
+  val minCreateOrUpdatePensionIncomeData: CreateUpdatePensionIncomeModel =
+    CreateUpdatePensionIncomeModel(minForeignPensionModel, minOverseasPensionContributionModel)
+  val minCreateOrUpdatePensionIncomeJsonBody: String = Json.toJson(minCreateOrUpdatePensionIncomeData).toString()
+
+
+  ".GetPensionIncomeConnector" should {
+    "include internal headers" when {
```
[the remark this answers is not in the document]
Comment (abiodxn) [reply it/connectors/PensionIncomeConnectorISpec.scala:110]: i think the only difference are the host names

**Comment by erasmos-hmrc** [comment]
>  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

Yes - however, I consider the ACs to be incomplete; for instance: there is no mention of the PUT and DELETE endpoints nor that fact that these are pass throughs - with the exception of the validation on the request body for the PUT.

As it stands, I don't believe that this story should have been started.

File: it/api/CreateOrAmendPensionIncomeITest.scala [hunk 13]
```diff
@@ -0,0 +1,253 @@
...
+        whenReady(buildClient(serviceUrl)
+          .withHttpHeaders(mtditidHeader)
+          .put(fullPensionIncomeJson)) {
+          result =>
+            result.status mustBe NO_CONTENT
+
+        }
+      }
+
+      "return 400 if the body payload validation fails" in new Setup {
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/api/CreateOrAmendPensionIncomeITest.scala]: Confirmed.

File: it/connectors/PensionIncomeConnectorISpec.scala [hunk 14]
```diff
@@ -0,0 +1,444 @@
...
+        stubPutWithoutResponseBody(desUrl, minCreateOrUpdatePensionIncomeJsonBody, NO_CONTENT)
+
+        implicit val hc: HeaderCarrier = HeaderCarrier()
+        val result = await(connector.createOrAmendPensionIncome(nino, taxYear, minCreateOrUpdatePensionIncomeData)(hc))
+
+        result mustBe Right(())
+      }
+    }
+
+    "return Left(error)" when {
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/connectors/PensionIncomeConnectorISpec.scala]: Confirmed.

File: it/connectors/PensionIncomeConnectorISpec.scala [hunk 15]
```diff
@@ -0,0 +1,444 @@
...
+        result mustBe Left(desError)
+      }
+
+    }
+  }
+
+  ".createOrAmendPensionIncome" should {
+
+    "include internal headers" when {
+      val headersSentToDes = Seq(
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/connectors/PensionIncomeConnectorISpec.scala]: Confirmed.

File: it/connectors/PensionIncomeConnectorISpec.scala [hunk 16]
```diff
@@ -0,0 +1,444 @@
...
+    "return a GetPensionIncomeModel when nino and taxYear are present" in {
+      val expectedResult = Json.parse(expectedResponseBody).as[GetPensionIncomeModel]
+      stubGetWithResponseBody(desUrl, OK, expectedResponseBody)
+
+      implicit val hc: HeaderCarrier = HeaderCarrier()
+      val result = await(connector.getPensionIncome(nino, taxYear)(hc)).right.get
+
+      result mustBe Some(expectedResult)
+    }
+
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/connectors/PensionIncomeConnectorISpec.scala:159]: No changes noticed. 

But this is entirely optional so wouldn't block my approval.

File: it/connectors/PensionIncomeConnectorISpec.scala (hunk 8 above)
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/connectors/PensionIncomeConnectorISpec.scala]: Confirmed.

[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/connectors/PensionIncomeConnectorISpec.scala]: I'm still getting this warning:

```
                                               Request was not matched
                                               =======================

-----------------------------------------------------------------------------------------------------------------------
| Closest stub                                             | Request                                                  |
-----------------------------------------------------------------------------------------------------------------------
                                                           |
POST                                                       | POST
/write/audit                                               | /write/audit/merged                                 <<<<< URL does not match
                                                           |
                                                           |
-----------------------------------------------------------------------------------------------------------------------
```

File: it/connectors/PensionIncomeConnectorISpec.scala (hunk 12 above)
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/connectors/PensionIncomeConnectorISpec.scala:118]: In that case the test name doesn't seem correct to me.

But it's a small point, so resolving.

File: it/connectors/PensionReliefsConnectorISpec.scala [hunk 17]
```diff
@@ -261,7 +261,7 @@ class PensionReliefsConnectorISpec extends WiremockSpec {
         implicit val hc: HeaderCarrier = HeaderCarrier(sessionId = Some(SessionId("sessionIdValue")))
         val connector = new PensionReliefsConnector(httpClient, appConfig(internalHost))
 
-        stubPutWithoutResponseBody(desUrl, fullCreateOrUpdatePensionReliefsJsonBody, NO_CONTENT)
+        stubPutWithoutResponseBody(desUrl, fullCreateOrUpdatePensionReliefsJsonBody, NO_CONTENT, headersSentToDes)
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/connectors/PensionReliefsConnectorISpec.scala:264]: For example, why do we not have the headers in "DES returns an unexpected error code - 502 BadGateway"?

File: it/connectors/PensionIncomeConnectorISpec.scala [hunk 18]
```diff
@@ -0,0 +1,444 @@
...
+import helpers.WiremockSpec
+import models._
+import play.api.Configuration
+import play.api.http.Status._
+import play.api.libs.json.Json
+import uk.gov.hmrc.http.{HeaderCarrier, HeaderNames, HttpClient, SessionId}
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+import utils.DESTaxYearHelper.desTaxYearConverter
+
+class PensionIncomeConnectorISpec extends WiremockSpec {
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/connectors/PensionIncomeConnectorISpec.scala:32]: No relevant changes spotted, but this was only a suggestion so certainly not a blocker.

Resolving.

File: it/connectors/PensionReliefsConnectorISpec.scala (hunk 17 above)
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/connectors/PensionReliefsConnectorISpec.scala:264]: Small point, so resolving.

File: app/services/PensionsService.scala [hunk 19]
```diff
@@ -56,4 +61,7 @@ class PensionsService @Inject()(reliefsConnector: PensionReliefsConnector,
 
   private def getStateBenefits(nino: String, taxYear: Int, mtditid: String)(implicit hc: HeaderCarrier): Future[GetStateBenefitsResponse] =
     stateBenefitsConnector.getStateBenefits(nino, taxYear)(hc.withExtraHeaders("mtditid" -> mtditid))
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply app/services/PensionsService.scala]: Confirmed.

File: app/services/PensionIncomeService.scala [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+import connectors.httpParsers.CreateOrAmendPensionIncomeHttpParser.CreateOrAmendPensionIncomeResponse
+import connectors.httpParsers.DeletePensionIncomeHttpParser.DeletePensionIncomeResponse
+import connectors.httpParsers.GetPensionIncomeHttpParser.GetPensionIncomeResponse
+import models.CreateUpdatePensionIncomeModel
+import uk.gov.hmrc.http.HeaderCarrier
+
+import javax.inject.Inject
+import scala.concurrent.Future
+
+class PensionIncomeService @Inject()(connector: PensionIncomeConnector) {
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply app/services/PensionIncomeService.scala:29]: Confirmed.

File: app/controllers/GetAllPensionsController.scala [hunk 21]
```diff
@@ -34,8 +34,8 @@ class GetAllPensionsController @Inject()(service: PensionsService,
 
   def getAllPensions(nino: String, taxYear: Int): Action[AnyContent] = auth.async { implicit user =>
     service.getAllPensionsData(nino, taxYear, user.mtditid).map {
-      case Right(AllPensionsData(pensionReliefs, pensionCharges, stateBenefits)) if pensionReliefs.isEmpty &&
-        pensionCharges.isEmpty && stateBenefits.isEmpty => NoContent
+      case Right(AllPensionsData(pensionReliefs, pensionCharges, stateBenefits, pensionIncome)) if pensionReliefs.isEmpty &&
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply app/controllers/GetAllPensionsController.scala]: Confirmed.

File: app/connectors/httpParsers/GetPensionIncomeHttpParser.scala [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+object GetPensionIncomeHttpParser extends DESParser {
+  type GetPensionIncomeResponse = Either[DesErrorModel, Option[GetPensionIncomeModel]]
+
+  override val parserName: String = "GetPensionIncomeHttpParser"
+
+  implicit object GetPensionIncomeHttpReads extends HttpReads[GetPensionIncomeResponse] {
+    override def read(method: String, url: String, response: HttpResponse): GetPensionIncomeResponse = {
+      response.status match {
+        case OK => response.json.validate[GetPensionIncomeModel].fold[GetPensionIncomeResponse](
+          jsonErrors => {
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply app/connectors/httpParsers/GetPensionIncomeHttpParser.scala]: Confirmed.

File: it/connectors/PensionIncomeConnectorISpec.scala (hunk 16 above)
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/connectors/PensionIncomeConnectorISpec.scala:159]: Resolving.

File: test/models/GetPensionIncomeModelSpec.scala [hunk 23]
```diff
@@ -0,0 +1,276 @@
...
+            amountBeforeTax = Some(1999.99),
+            taxTakenOff = Some(1999.99),
+            specialWithholdingTax = Some(1999.99),
+            foreignTaxCreditRelief = Some(false)
+          )
+        ),
+        overseasPensionContribution = Seq(
+          OverseasPensionContribution(
+            customerReference = None,
+            exemptEmployersPensionContribs = 1999.99,
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply test/models/GetPensionIncomeModelSpec.scala:230]: I confirmed with @abiodxn that this is indeed realistic.

File: it/connectors/PensionReliefsConnectorISpec.scala [hunk 24]
```diff
@@ -272,7 +272,7 @@ class PensionReliefsConnectorISpec extends WiremockSpec {
         implicit val hc: HeaderCarrier = HeaderCarrier(sessionId = Some(SessionId("sessionIdValue")))
         val connector = new PensionReliefsConnector(httpClient, appConfig(externalHost))
 
-        stubPutWithoutResponseBody(desUrl, fullCreateOrUpdatePensionReliefsJsonBody, NO_CONTENT)
+        stubPutWithoutResponseBody(desUrl, fullCreateOrUpdatePensionReliefsJsonBody, NO_CONTENT, headersSentToDes)
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/connectors/PensionReliefsConnectorISpec.scala:275]: Small point, so resolving.

File: app/services/PensionIncomeService.scala [hunk 25]
```diff
@@ -0,0 +1,43 @@
...
+
+  def getPensionIncome(nino: String, taxYear: Int)(implicit hc: HeaderCarrier): Future[GetPensionIncomeResponse] =
+    connector.getPensionIncome(nino, taxYear)
+
+    def createOrAmendPensionIncome(nino: String, taxYear: Int,
+                                    pensionIncome: CreateUpdatePensionIncomeModel)
+                                   (implicit hc: HeaderCarrier): Future[CreateOrAmendPensionIncomeResponse] = {
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply app/services/PensionIncomeService.scala]: Confirmed.

File: it/connectors/PensionIncomeConnectorISpec.scala (hunk 18 above)
[the remark this answers is not in the document]
Comment (abiodxn) [reply it/connectors/PensionIncomeConnectorISpec.scala:32]: I did make changes and reduce the duplication. I now have two sequences for errors in this spec for the `.GetPensionIncomeConnector.handle error` scenario

File: it/connectors/PensionIncomeConnectorISpec.scala (hunk 8 above)
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/connectors/PensionIncomeConnectorISpec.scala]: I see that the fix was made.

However, there's one scenario where the `auditStubs` call is missing: "return a NOT_FOUND" (around line 160).

File: it/connectors/PensionIncomeConnectorISpec.scala (hunk 18 above)
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/connectors/PensionIncomeConnectorISpec.scala:32]: OK

File: it/connectors/PensionIncomeConnectorISpec.scala (hunk 8 above)
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/connectors/PensionIncomeConnectorISpec.scala]: Confirmed.

**Review by erasmos-hmrc — APPROVED** [review]

**Comment by erasmos-hmrc** [comment]
Sorry, @abiodxn  - I haven't run the journey tests, yet.

Will do so now.

**Comment by erasmos-hmrc** [comment]
Journey tests successfully run:
```
[info] ScalaTest
[info] Run completed in 32 minutes, 36 seconds.
[info] Total number of tests run: 0
[info] Suites: completed 0, aborted 0
[info] Tests: succeeded 0, failed 0, canceled 0, ignored 0, pending 0
[info] No tests were executed.
[info] Passed: Total 82, Failed 0, Errors 0, Passed 82
[success] Total time: 1961 s (32:41), completed 5 Aug 2022, 17:32:24
```

Note that I started this service with:

```
sbt -Dplay.akka.http.server.request-timeout=90s run
```

Which may not have been necessary, but just in case.

**Pull request merged**